src/BorrowLogic/BorrowHandlers.sol

```diff
@@ -28,19 +27,10 @@ abstract contract BorrowHandlers is IBorrowHandlers, BorrowCheckers, SafeMint {
     function useOffer(
         OfferArg memory arg,
         CollateralState memory collatState
-    ) internal returns (CollateralState memory) {
+    ) internal view returns (CollateralState memory, address /* signer */) {
         address signer = checkOfferArg(arg);
         Ray shareMatched;
 
-        if (arg.offer.assetToLend != collatState.assetLent) {
-            // all offers used for a collateral must refer to the same erc20
-            revert InconsistentAssetRequests(collatState.assetLent, arg.offer.assetToLend);
-        }
-        if (arg.offer.tranche != collatState.tranche) {
-            // all offers used for a collateral must refer to the same interest rate tranche
-            revert InconsistentTranches(collatState.tranche, arg.offer.tranche);
-        }
-
```

Comment:
This checks that assets and interest rates are consistent across offers but is no longer needed since users are now limited to one offer per loan